scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/CascadingBackend.scala

```diff
@@ -149,21 +154,18 @@ object CascadingBackend {
       RichPipe.setPipeDescriptions(p, ordered ::: unordered)
     }
 
-    def loop[T](t: TypedPipe[T], rest: FlatMappedFn[T, U], descriptions: List[(String, Boolean)]): Pipe = t match {
-      case cp@CascadingPipe(_, fields, localFlowDef, m, _) =>
-        import Dsl.flowDefToRichFlowDef
-        // This check is not likely to fail unless someone does something really strange.
-        // for historical reasons, it is not checked by the typed system
-        require(m == mode,
-          s"Cannot switch Mode between TypedSource.read and toPipe calls. Pipe: $p, pipe mode: $m, outer mode: $mode")
-        flowDef.mergeFrom(localFlowDef)
-
-        def go[T1 <: T](cp: CascadingPipe[T1]): Pipe = {
-          val withRest = RichPipe(cp.pipe).flatMapTo[T1, U](fields -> fieldNames)(rest)(cp.converter, setter)
-          applyDescriptions(withRest, descriptions)
-        }
-        go(cp)
+    def finish[T](cp: CascadingPipe[T],
```

Comment:
might be worth a quick comment on this and loop below just to say what they do.

This i think means we merge our new flow def from the cascading pipe with an historical one, flatMap with the function we kept inside the pipe, finally applying the descriptions on the cascading pipe.